### PR TITLE
Set initial FPU state and mask SIMD FP exceptions

### PR DESCRIPTION
**Pull request by conradgrobler**
The Oak Restricted Kernel does not provide custom floating-point exception handlers to resolve problems that might occur, so we mask the exceptions so that we fall back on the default behaviour so that we don't shut down the entire VM due to a loss of precision or underflow when an application does floating point arithmetic.

### CODE REVIEW

**Comment by tiziano88** [comment]
Is it possible to have a test that triggers the exception (and others too perhaps)?

**Comment by conradgrobler** [comment]
> Is it possible to have a test that triggers the exception (and others too perhaps)?

It is not trivial, since these are only modified when booting the restricted kernel inside a VM. We would have to create a separate enclave application with a separate launcher that will run it as part of the kokoro tests.

**Comment by conradgrobler** [comment]
Created https://github.com/project-oak/oak/issues/4402 to track this.

File: oak_restricted_kernel/src/avx.rs [hunk 1]
```diff
@@ -36,5 +41,16 @@ pub fn enable_avx() {
         let mut xcr0 = XCr0::read();
         xcr0 |= XCr0Flags::X87 | XCr0Flags::SSE | XCr0Flags::AVX;
         XCr0::write(xcr0);
+        // Initialize x87 FPU to a default state.
+        asm!("fninit");
+        // Mask SIMD floating-point exceptions to enable default behavior.
+        let mut mxcsr = read();
+        mxcsr.insert(MxCsr::DENORMAL_MASK);
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply oak_restricted_kernel/src/avx.rs]: Done.

**Comment by conradgrobler** [comment]
> I'll have to admit not knowing much about this -- is masking exceptions like this by default normal? What does Linux do by default, for example?

Linux mask all the exceptions by default, e.g. when waking up a new AP in SEV it sets mxcsr to a default value where all these are masked: https://github.com/torvalds/linux/blob/4f82870119a46b0d04d91ef4697ac4977a255a9d/arch/x86/kernel/sev.c#L60

**Pull request merged**